// Copyright Â© 2017-2021 Trust Wallet.
//
// This file is part of Trust. The full Trust copyright notice, including
// terms governing use, modification, and redistribution, is contained in the
// file LICENSE at the root of the source code distribution tree.

syntax = "proto3";

package TW.Top.Proto;
option java_package = "wallet.core.jni.proto";

// TODO: Input data necessary to create a signed transaction.
message SigningInput {
    string from = 1;
    string to = 2;
    uint32 tx_type = 3;
    uint64 amount = 4;
    uint32 extra = 5;
    uint32 tx_deposit = 6;
    uint32 source_action_type = 7;
    uint32 target_action_type = 8;
    uint64 last_tx_nonce = 9;
    string last_tx_hash = 10;
    string note = 11;
    // Private key.
    bytes private_key = 12;
}

// Transaction signing output.
message SigningOutput {
    // Signed and encoded transaction bytes.
    bytes encoded = 1;

    // Signature.
    bytes signature = 2;
}
